test.py

```diff
@@ -189,16 +225,24 @@ def layernorm_bwd_batcher(
 ) -> tuple[Sequence[jax.Array], Sequence[int | None]]:
   dz, x, mu, rsigma, gamma = batched_args
   dz_bdim, x_bdim, mu_bdim, rsigma_bdim, gamma_bdim = batch_dims
-  if not (rsigma_bdim is gamma_bdim is batching.not_mapped):
-    raise NotImplementedError
-  if x_bdim == x.ndim - 1:
-    x = jnp.moveaxis(x, -1, -2)
-    x_bdim = x.ndim - 2
-  out_bdims = dz_bdim, x_bdim, batching.not_mapped
-  return layernorm_bwd_p.bind(dz, x, mu, rsigma, gamma, zero_centered_gamma=zero_centered_gamma, epsilon=epsilon), out_bdims
+  if gamma_bdim is not not_mapped:
+    raise NotImplementedError  # kernel doesn't support gamma batching
+  if x_bdim != mu_bdim or x_bdim != rsigma_bdim:
+    raise NotImplementedError  # TODO: just jnp.moveaxis/broadcast them to match
```

Comment:
The user should never be able to hit these cases (as we wrote the calling code here).